src/util/createInstantSearchComponent.js

```diff
@@ -0,0 +1,140 @@
+import instantsearch from 'instantsearch.js/es/';
+import { createSuitMixin } from '../mixins/suit';
+import { warn } from './warn';
+import { version } from '../../package.json'; // rollup does pick only what needed from json
+
+const oldApi = () =>
+  warn(
+    `Vue InstantSearch: You used the prop api-key or api-key.
+These have been replaced by search-client.
+
+See more info here: https://community.algolia.com/vue-instantsearch/components/InstantSearch.html#usage`
+  );
+
+export const createInstantSearchComponent = ({ ssr }) => ({
+  name: ssr ? 'AisInstantSearchSsr' : 'AisInstantSearch',
```

Comment:
Rather than conditions that are a bit hard to read we could move each configuration that is dedicated to a specific instance inside his own file and keep all the share logic here. The function could accept a subset of the declaration and merge it to create the declaration of the component.

```js
createInstantSearchComponent({
  data() {
    return {
      instantsearchInstance: instantsearch({ ... }),
    },
  }
})
```